ddtrace/tracer/textmap.go

```diff
@@ -114,74 +115,188 @@ func NewPropagator(cfg *PropagatorConfig) Propagator {
 	if cfg.PriorityHeader == "" {
 		cfg.PriorityHeader = DefaultPriorityHeader
 	}
-	return &propagator{cfg}
+	return &propagator{
+		injectors:  makeInjectors(cfg),
+		extractors: makeExtractors(cfg),
+	}
 }
 
-// propagator implements a propagator which uses TextMap internally.
+// propagator implements a Propagator that supports TextMap carriers.
 // It propagates the trace and span IDs, as well as the baggage from the
 // context.
-type propagator struct{ cfg *PropagatorConfig }
+type propagator struct {
+	injectors  []textMapPropagator
+	extractors []textMapPropagator
+}
+
+// makeInjectors returns a list of injectors to apply when propagating a
+// context. By default, only the datadog propagation style will be used.
+// If the DD_PROPAGATION_STYLE_INJECT environment variable is set,
+// this can override the default behavior.
+func makeInjectors(cfg *PropagatorConfig) []textMapPropagator {
+	dd := &datadogPropagator{cfg}
+	b3 := &b3Propagator{}
+
+	ps := os.Getenv("DD_PROPAGATION_STYLE_INJECT")
+	if ps == "" {
+		return []textMapPropagator{dd}
+	}
+	styles := propagationStyleList(ps)
+
+	var injectors []textMapPropagator
+	for _, v := range styles {
+		switch v {
+		case "Datadog":
+			injectors = append(injectors, dd)
+		case "B3":
+			injectors = append(injectors, b3)
+		default:
+			// TODO: consider logging something for invalid/unknown styles.
+		}
+	}
+
+	// If all the styles were invalid/unknown, then revert to default behavior.
```

Comment:
One personal style I've used throughout the repository (and in general) is to always put comments referring to the outcome of an if clause, inside the "if" clause. It would be nice if you could do the same to keep consistency, e.g.:
```go
if len(injectors) == 0 {
    // ensure that at least the Datadog propagator is present
    ...
}
```